File: Dabonda/app/src/main/java/com/creator/dabonda/adapters/CustomImageSelectAdapter.java
package com.creator.dabonda.adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.creator.dabonda.R;
import com.creator.dabonda.models.Image;

import java.util.ArrayList;

/**
 * Created by Darshan on 4/18/2015.
 */
public class CustomImageSelectAdapter extends CustomGenericAdapter<Image> {
    private boolean mIsSelectedCheck = true;

    public CustomImageSelectAdapter(Context context, ArrayList<Image> images) {
        super(context, images);
    }

    public CustomImageSelectAdapter(Context context, ArrayList<Image> images, boolean isSelectedCheck) {
        super(context, images);

        mIsSelectedCheck = isSelectedCheck;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder viewHolder;

        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.grid_view_item_image_select, null);

            viewHolder = new ViewHolder();
            viewHolder.imageView = (ImageView) convertView.findViewById(R.id.image_view_image_select);
            viewHolder.textView = (TextView) convertView.findViewById(R.id.image_view_image_sequence);
            viewHolder.view = convertView.findViewById(R.id.view_alpha);

            convertView.setTag(viewHolder);

        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }

        viewHolder.imageView.getLayoutParams().width = size;
        viewHolder.imageView.getLayoutParams().height = size;

        viewHolder.view.getLayoutParams().width = size;
        viewHolder.view.getLayoutParams().height = size;

        if (mIsSelectedCheck && arrayList.get(position).isSelected) {
            viewHolder.textView.setVisibility(View.VISIBLE);
            viewHolder.textView.setText(arrayList.get(position).sequence + "");

            viewHolder.view.setAlpha(0.5f);
            ((FrameLayout) convertView).setForeground(context.getResources().getDrawable(R.drawable.ic_done_white));
        } else {
            viewHolder.textView.setVisibility(View.INVISIBLE);

            viewHolder.view.setAlpha(0.0f);
            ((FrameLayout) convertView).setForeground(null);
        }

        Glide.with(context)
                .load(arrayList.get(position).path)
                .placeholder(R.drawable.image_placeholder).into(viewHolder.imageView);

        return convertView;
    }

    private static class ViewHolder {
        public ImageView imageView;
        public TextView textView;
        public View view;
    }
}

File: 2023/PriceHunter/app/src/main/java/com/creator/pricehunter/adapters/CustomImageSelectAdapter.kt
package com.creator.pricehunter.adapters

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.creator.pricehunter.R
import com.creator.pricehunter.models.Image

/**
 * Created by Darshan on 4/18/2015.
 */
class CustomImageSelectAdapter : CustomGenericAdapter<Image> {
    private var mIsSelectedCheck = true

    constructor(context: Context?, images: ArrayList<Image>?) : super(context, images) {}
    constructor(context: Context?, images: ArrayList<Image>?, isSelectedCheck: Boolean) : super(
        context,
        images
    ) {
        mIsSelectedCheck = isSelectedCheck
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var convertView = convertView
        val viewHolder: ViewHolder
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.grid_view_item_image_select, null)
            viewHolder = ViewHolder()
            viewHolder.imageView =
                convertView.findViewById<View>(R.id.image_view_image_select) as ImageView
            viewHolder.textView =
                convertView.findViewById<View>(R.id.image_view_image_sequence) as TextView
            viewHolder.view = convertView.findViewById(R.id.view_alpha)
            convertView.tag = viewHolder
        } else {
            viewHolder = convertView.tag as ViewHolder
        }
        viewHolder.imageView!!.layoutParams.width = size
        viewHolder.imageView!!.layoutParams.height = size
        viewHolder.view!!.layoutParams.width = size
        viewHolder.view!!.layoutParams.height = size
        if (mIsSelectedCheck && arrayList!![position]!!.isSelected) {
            viewHolder.textView!!.visibility = View.VISIBLE
            viewHolder.textView!!.text = arrayList!![position]!!.sequence.toString() + ""
            viewHolder.view!!.alpha = 0.5f
            (convertView as FrameLayout).foreground =
                context!!.resources.getDrawable(R.drawable.ic_done_white)
        } else {
            viewHolder.textView!!.visibility = View.INVISIBLE
            viewHolder.view!!.alpha = 0.0f
            (convertView as FrameLayout).foreground = null
        }
        Glide.with(context)
            .load(arrayList!![position]!!.path)
            .placeholder(R.drawable.image_placeholder).into(viewHolder.imageView)
        return convertView
    }

    private class ViewHolder {
        var imageView: ImageView? = null
        var textView: TextView? = null
        var view: View? = null
    }
}